actionpack/lib/action_dispatch/middleware/executor.rb

```diff
@@ -15,7 +15,9 @@ def call(env)
       begin
         response = @app.call(env)
         if rendered_error = env["action_dispatch.exception"]
-          @executor.error_reporter.report(rendered_error, handled: false, source: "application.action_dispatch")
+          unless ExceptionWrapper.rescue_response?(rendered_error)
```

Comment:
Instead of doing this here, why we just don't add it to `env["action_dispatch.exception"]` if it isn't rescuable?

Reply:
We can't do that. But we can add the exception to a different key in the env only if we want it to be reported.

Reply:
I agree, my biggest concern was the use of `ExceptionWrapper.rescue_response?` in the executor. Feels like executor shouldn't ideally know about the wrapper.

>We can't do that

Just to make sure we have the same understanding. Is that because `exceptions_app` (`ActionDispatch::PublicExceptions` by default) expects `action_dispatch.exception` header to be present?
What if we still pass the header to `exceptions_app` but drop it afterwards? 

I guess using a different key makes more sense, I just wasn't sure if it's okay to add more headers. Let me try that!

Reply:
So I added a `"action_dispatch.report_exception"` boolean flag to indicate whether the exception should be reported or not. I felt like it's excessive to have two headers pointing to the same exception in case if it needs to be reported.